django_log_formatter_asim/__init__.py

```diff
@@ -1,176 +1,127 @@
 import json
 import logging
-import os
-import platform
-from urllib.parse import urlparse
+from datetime import datetime
+from importlib.metadata import distribution
 
 from django.conf import settings
-# TODO...
-from kubi_ecs_logger import Logger
-from kubi_ecs_logger.models import BaseSchema
-from kubi_ecs_logger.models import Severity
-
-# TODO: Event categories - https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html  # noqa E501
-CATEGORY_DATABASE = "database"
-CATEGORY_PROCESS = "process"
-CATEGORY_WEB = "web"
-
-
-class ASIMLogger(Logger):
-    def __init__(self, *args, **kwargs):
-        super(ASIMLogger, self).__init__(*args, **kwargs)
-
-    def get_log_dict(self):
-        return BaseSchema().dump(self._base)
 
 
 class ASIMFormatterBase:
     def __init__(self, record):
         self.record = record
 
-    def _get_event_base(self, extra_labels={}):
-        labels = {
-            "application": getattr(settings, "DLFE_APP_NAME", None),
-            "env": self._get_environment(),
+        if not getattr(settings, "DLFA_LOG_PERSONALLY_IDENTIFIABLE_INFORMATION", False):
+            self._replace_personally_identifiable_information()
+
+    def _replace_personally_identifiable_information(self):
+        if getattr(self.record, "request", None):
+            user = getattr(self.record.request, "user", None)
+            if user:
+                user.username = "{{USERNAME}}"
+                user.email = "{{EMAIL}}"
+                user.first_name = "{{FIRST_NAME}}"
+                user.last_name = "{{LAST_NAME}}"
+
+    def _get_log_dict_base(self):
+        record = self.record
+        log_time = datetime.utcfromtimestamp(record.created).isoformat()
+        log_dict = {
+            # Event fields...
+            "EventMessage": record.msg,
+            "EventCount": 1,
+            "EventStartTime": log_time,
+            "EventEndTime": log_time,
+            "EventType": record.name,
+            "EventResult": "NA",
+            "EventSeverity": self._get_event_severity(record.levelname),
+            "EventOriginalSeverity": record.levelname,
+            "EventSchema": "ProcessEvent",
+            "EventSchemaVersion": "0.1.4",
+            "ActingAppType": "Django",
+            # Other fields...
+            "AdditionalFields": {
+                "DjangoLogFormatterAsimVersion": distribution("django-log-formatter-asim").version,
```

Comment:
Seemed like it would be good to be able to identify applications where we want to upgrade this package.